plugin/src/node-builder.ts

```diff
@@ -53,43 +53,73 @@ const downloadImageAndCreateFileNode = async (
   return undefined;
 };
 
+interface ProcessorMap {
+  [remoteType: string]: (
+    node: NodeInput,
+    gatsbyApi: SourceNodesArgs,
+    options: ShopifyPluginOptions
+  ) => Promise<void>;
+}
+
+const processorMap: ProcessorMap = {
+  ShopifyLineItem: async (node) => {
+    const lineItem = node;
+    lineItem.productId = (lineItem.product as { id: string }).id;
+    delete lineItem.product;
+  },
+  ProductImage: async (node, gatsbyApi, options) => {
+    if (options.downloadImages) {
+      const url = node.originalSrc as string;
+      const fileNodeId = await downloadImageAndCreateFileNode(
+        {
+          url,
+          nodeId: node.id,
+        },
+        gatsbyApi
+      );
+
+      node.localFile = fileNodeId;
+    }
+  },
+  Product: async (node, gatsbyApi, options) => {
+    if (options.downloadImages) {
+      const featuredImage = node.featuredImage as
+        | {
+            originalSrc: string;
+            localFile: string | undefined;
+          }
+        | undefined;
+
+      if (featuredImage) {
+        const url = featuredImage.originalSrc;
+        const fileNodeId = await downloadImageAndCreateFileNode(
+          {
+            url,
+            nodeId: node.id,
+          },
+          gatsbyApi
+        );
+
+        featuredImage.localFile = fileNodeId;
```

Comment:
Just realized this link is missing from the schema, will have to update that for this to work correctly